### DISCUSSION TITLE
Why is it not possible to change the replica count in v2 longhorn volume?

### DISCUSSION THREAD

**Discussion by hookak** [longhorn/longhorn#9804] [Q&A]
Hello,
I'd like to manually adjust the replica count for each v2 LHV, but it prevented by a [wehbook](https://github.com/longhorn/longhorn-manager/blob/70ac3bcf10ab37e05bbb6c5d23d4bbf89794d16f/webhook/resources/volume/validator.go#L254)
To change the replica count for v2 LHV is already possible during node eviction. 
The replica count of v2 LHV changes in the sequence n->n+1->n.
What is the reason for this restriction?

**Comment by derekbit** [comment]
> Hello, I'd like to manually adjust the replica count for each v2 LHV, but it prevented by a [wehbook](https://github.com/longhorn/longhorn-manager/blob/70ac3bcf10ab37e05bbb6c5d23d4bbf89794d16f/webhook/resources/volume/validator.go#L254) To change the replica count for v2 LHV is already possible during node eviction. The replica count of v2 LHV changes in the sequence n->n+1->n. What is the reason for this restriction?

v1.7.2?

**Reply by hookak** [reply]
I tested on v1.7.0, but even in v1.7.2, the functionality is still restricted in volume validation.

**Reply by derekbit** [reply]
OK. We don't remove the validation in v1.7.2. The validation should be removed.
```
	if types.IsDataEngineV2(newVolume.Spec.DataEngine) {
		// TODO: remove this check when we support the following features for SPDK volumes
		if oldVolume.Spec.Size != newVolume.Spec.Size {
			err := fmt.Errorf("changing volume size for volume %v is not supported for data engine %v",
				newVolume.Name, newVolume.Spec.DataEngine)
			return werror.NewInvalidError(err.Error(), "")
		}

		if oldVolume.Spec.NumberOfReplicas != newVolume.Spec.NumberOfReplicas {
			err := fmt.Errorf("changing number of replicas for volume %v is not supported for data engine %v",
				newVolume.Name, newVolume.Spec.DataEngine)
			return werror.NewInvalidError(err.Error(), "")
		}
```

@hookak Would you like to contribute a PR for fixing the issue? Thank you.

**Reply by derekbit** [reply]
cc @shuo-wu

**Reply by hookak** [reply]
@derekbit Of course. I'm always grateful for the improved usability of the v2 volume.
